### PR TITLE
Typescript: fix: add paramPropertyNaming

### PR DESCRIPTION
**Pull request by mfmarche**
Model naming conforms to the property: modelPropertyNaming, however the
parameter naming scheme is currently hardcoded to camelcase. Since the
parameter names are closely tied to the model properties names, is maybe
desirable to use a similar naming approach.

The parameter naming cannot directly use the modelPropertyNaming since that
will cause issues with current generated code, since the
modelPropertyNaming currently defaults to "original"

A new property has been created to override this behavior, named
"paramPropertyNaming", and its default is set to "camelcase".

This PR likely addresses: https://github.com/OpenAPITools/openapi-generator/issues/8296 

### PR checklist
 
- [x ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
cc @TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02)

File: modules/openapi-generator/src/main/resources/python/model_utils.mustache [hunk 1]
```diff
@@ -1450,7 +1450,12 @@ def get_allof_instances(self, model_args, constant_args):
     for allof_class in self._composed_schemas['allOf']:
 
         try:
-            allof_instance = allof_class(**model_args, **constant_args)
+            # FIX/workaround.
+            # get_allof_instances could be called from from_openapi_data or __init__
+            # of class. If it is called from _from_openapi_data, then we must not check
+            # for readOnly validation. However this method has no knowledge of the caller,
+            # so for now, use _from_openapi_data to instantiate the object.
+            allof_instance = allof_class._from_openapi_data(**model_args, **constant_args)
```
Comment (amakhrov) [inline modules/openapi-generator/src/main/resources/python/model_utils.mustache:1458]: this looks like an unrelated change - should it be reverted here?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenConstants.java [hunk 2]
```diff
@@ -205,12 +205,17 @@
     public static final String MODEL_PROPERTY_NAMING = "modelPropertyNaming";
     public static final String MODEL_PROPERTY_NAMING_DESC = "Naming convention for the property: 'camelCase', 'PascalCase', 'snake_case' and 'original', which keeps the original name";
 
+    public static final String PARAM_PROPERTY_NAMING = "paramPropertyNaming";
```
Comment (amakhrov) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenConstants.java:208]: the name sounds ambiguous - like it's naming for properties of parameters. but parameters do not have properties.
maybe just `param naming` instead?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java [hunk 3]
```diff
@@ -212,6 +214,10 @@ public void processOpts() {
             setModelPropertyNaming((String) additionalProperties.get(CodegenConstants.MODEL_PROPERTY_NAMING));
         }
 
+        if (additionalProperties.containsKey(CodegenConstants.PARAM_PROPERTY_NAMING)) {
```
Comment (amakhrov) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java:217]: I think it should be made a part of cli options - which would automatically add it to documentation. Otherwise nobody will ever know about it.

See how other options are added in the constructor: https://github.com/OpenAPITools/openapi-generator/blob/eb28082c6187d638b590c0dc75adce78a1153444/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java#L176

**Review by amakhrov — COMMENTED** [review]
Thanks for your contribution!
Overall looks good to me. Just a few minor notes.

I believe existing samples are not effected, since the default value is backward compatible.
Could you please add some unit tests that would verify the new setting works as expected?

File: modules/openapi-generator/src/main/resources/python/model_utils.mustache (hunk 1 above)
Comment (mfmarche) [reply modules/openapi-generator/src/main/resources/python/model_utils.mustache:1458]: yes i will revert, that is another local fix related to another issue I found.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java (hunk 3 above)
Comment (mfmarche) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java:217]: thanks, I will make that change.

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptnode/TypeScriptNodeClientOptionsTest.java [hunk 4]
```diff
@@ -42,6 +42,7 @@ protected CodegenConfig getCodegenConfig() {
     protected void verifyOptions() {
         verify(clientCodegen).setSortParamsByRequiredFlag(Boolean.valueOf(TypeScriptNodeClientOptionsProvider.SORT_PARAMS_VALUE));
         verify(clientCodegen).setModelPropertyNaming(TypeScriptNodeClientOptionsProvider.MODEL_PROPERTY_NAMING_VALUE);
+        verify(clientCodegen).setParamNaming(TypeScriptNodeClientOptionsProvider.PARAM_NAMING_VALUE);
```
Comment (amakhrov) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptnode/TypeScriptNodeClientOptionsTest.java:45]: nice!

but I also meant a regular unit test that checks the logic behind the new option, not just that it's properly supported by cli interface.

e.g. like this tests makes sure enumPropertyNaming is honored when formatting enums: https://github.com/OpenAPITools/openapi-generator/blob/2364dd7eb18d6154a89f5f638e88ba7fbbaa684e/modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptangular/TypeScriptAngularClientCodegenTest.java#L71

Comment (mfmarche) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/typescript/typescriptnode/TypeScriptNodeClientOptionsTest.java:45]: thanks for the pointer, I updated the test accordingly.

**Review by amakhrov — APPROVED** [review]
Looks great!

**Comment by mfmarche** [comment]
Hi @amakhrov thanks for the review. Is there something I need to do next to move it along?
